app/models/component.rb

```diff
@@ -75,4 +74,24 @@ def type
   def display
     I18n.t("user_journey.component_name.#{type}")
   end
+
+private
+
+  def encryption_certificate_check
+    x509 = encryption_certificate&.x509
```

Comment:
This is very good as usual, I thought this might be better

In Certificate model
```
  def expires_soon?
    x509.not_after - Time.now < 30.day
  end
```
In Component model
```
  def encryption_certificate_check
    if encryption_certificate&.expired?
      Rails.logger.error "When publishing the meta data the service '#{name}' has been identified as having an expired encryption certificate."
      nil
    else
      encryption_certificate
    end
  end

```

Reply:
I'll meet you half way... I've changed the method names but I am leaving everything else alone... So please approve.